classes/shop/Shop.php

```diff
@@ -369,7 +369,7 @@ public static function initialize()
                         $redirect_header = ($redirect_type == 1 ? 'Found' : 'Moved Permanently');
                         header('HTTP/1.0 ' . $redirect_code . ' ' . $redirect_header);
                         header('Cache-Control: no-cache');
-                        header('Location: http://' . $url);
+                        header('Location: ' . Tools::getShopProtocol() . $url);
```

Comment:
This one by testing a wrong domain.

For instance, if you installed your shop locally with the domain `127.0.0.1`, this code will be called by reaching your website from `http(s)://localhost`.